src/pdn/src/grid.cpp

```diff
@@ -1095,22 +1095,35 @@ const odb::Rect InstanceGrid::getDomainBoundary() const
 
 const odb::Rect InstanceGrid::getGridArea() const
 {
-  return applyHalo(getDomainArea());
+  return applyHalo(getDomainArea(), false, true, true);
 }
 
-odb::Rect InstanceGrid::applyHalo(const odb::Rect& rect) const
+odb::Rect InstanceGrid::applyHalo(const odb::Rect& rect,
+                                  bool rect_is_min,
+                                  bool apply_horizontal,
+                                  bool apply_vertical) const
 {
-  return applyHalo(rect, halos_);
+  return applyHalo(rect, halos_, rect_is_min, apply_horizontal, apply_vertical);
 }
 
-odb::Rect InstanceGrid::applyHalo(const odb::Rect& rect, const InstanceGrid::Halo& halo)
+odb::Rect InstanceGrid::applyHalo(const odb::Rect& rect,
```

Comment:
I'm not seeing how this prevents negative halos.  Would you explain a bit?

Reply:
`rect_is_min` if true, will merge the source rect back into the shape. Causing the negative halo to be "removed"